packages/smart-vaults/src/library/MultiSignerLib.sol

```diff
@@ -3,38 +3,31 @@ pragma solidity ^0.8.23;
 
 import { decodeAccountSigner } from "../signers/AccountSigner.sol";
 import { decodePasskeySigner } from "../signers/PasskeySigner.sol";
+import { Signer } from "../signers/Signer.sol";
+
+/// @notice Storage layout used by this contract.
+/// @dev Can allow up to 256 signers.
```

Comment:
[According to ERC-7201](https://eips.ethereum.org/EIPS/eip-7201#customstorage-location), the annotation `@custom:storage-location` below this line is ignored when outside of a contract:

> Structs with this annotation found outside of contracts are not considered to be namespaces for any contract in the source code.

(Inside the library like it was before it would also be ignored.)

The practical consequence is that tools like OpenZeppelin Upgrades will not consider this struct when analyzing the storage layout of a vault.

A good way to fix this IMO would be to keep the struct here but remove the annotation, and define a new struct specifically for the layout inside the `MultiSigner` contract.

The names may need to be reviewed. I see this library as another kind of signer so it may belong in `src/signers/MultiSigner.sol`. As for the contract it could be `contract MultiSignerAuth`?

```solidity
contract MultiSignerAuth {
    /// @custom:storage-location erc7201:splits.storage.MultiSigner
    struct MultiSignerAuthStorage {
        MultiSigner signer;
    }

    ...
}
```